Shared/Samples/Geocode offline/GeocodeOfflineView.swift

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct GeocodeOfflineView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The viewpoint of the map view, initially centered on San Diego, CA, USA.
+    @State private var viewpoint = Viewpoint(
+        center: Point(x: -13_042_250, y: 3_857_970, spatialReference: .webMercator),
+        scale: 2e4
+    )
+    
+    /// The text in the search bar.
+    @State private var searchText = ""
+    
+    /// The search text that has been submitted.
+    @State private var submittedSearchText: String?
+    
+    /// A Boolean value indicating whether the "No results found." alert is showing.
+    @State private var isShowingResultAlert = false
+    
+    /// A pre-populated list of example addresses.
+    private let exampleAddresses = [
+        "910 N Harbor Dr, San Diego, CA 92101",
+        "2920 Zoo Dr, San Diego, CA 92101",
+        "111 W Harbor Dr, San Diego, CA 92101",
+        "868 4th Ave, San Diego, CA 92101",
+        "750 A St, San Diego, CA 92101"
+    ]
+    
+    var body: some View {
+        GeocodeMapView(model: model, viewpoint: $viewpoint)
```

Comment:
you can also pass the model through the environment instead of explicitly in the init.